apps/system/js/app_window.js

```diff
@@ -1830,33 +1844,42 @@
     }
 
     this.debug('requesting to open');
+
+    // The homescreen was scheduled for a setVisible(false) but we need it
+    // right away.
+    if (this.isHomescreen && this._visibilityChangeTimeout) {
+      this.setVisible(true);
+      callback();
```

Comment:
Sometimes this callback is called in a setTimeout, other times it's called immediately - do we want/need to be consistent on this?